runtime/wasm/src/host.rs

```diff
@@ -108,82 +92,43 @@ where
         };
 
         RuntimeHost::new(
-            logger,
             ethereum_adapter.clone(),
             self.link_resolver.clone(),
             store.clone(),
+            store.clone(),
             RuntimeHostConfig {
                 subgraph_id,
                 mapping: data_source.mapping,
                 data_source_name: data_source.name,
                 contract: data_source.source,
                 templates,
             },
+            mapping_request_sender,
         )
     }
 }
 
-type MappingResponse = (Result<BlockState, Error>, futures::Finished<Instant, Error>);
-
-#[derive(Debug)]
-struct MappingRequest {
-    logger: Logger,
-    block: Arc<EthereumBlock>,
-    state: BlockState,
-    trigger: MappingTrigger,
-    result_sender: oneshot::Sender<MappingResponse>,
-}
-
 #[derive(Debug)]
-enum MappingTrigger {
-    Log {
-        transaction: Arc<Transaction>,
-        log: Arc<Log>,
-        params: Vec<LogParam>,
-        handler: MappingEventHandler,
-    },
-    Call {
-        transaction: Arc<Transaction>,
-        call: Arc<EthereumCall>,
-        inputs: Vec<LogParam>,
-        outputs: Vec<LogParam>,
-        handler: MappingCallHandler,
-    },
-    Block {
-        handler: MappingBlockHandler,
-    },
-}
-
-#[derive(Debug)]
-pub struct RuntimeHost {
+struct RuntimeHost {
     data_source_name: String,
     data_source_contract: Source,
     data_source_contract_abi: MappingABI,
     data_source_event_handlers: Vec<MappingEventHandler>,
     data_source_call_handlers: Vec<MappingCallHandler>,
     data_source_block_handlers: Vec<MappingBlockHandler>,
     mapping_request_sender: Sender<MappingRequest>,
-    _guard: oneshot::Sender<()>,
+    host_exports: Arc<HostExports>,
 }
 
 impl RuntimeHost {
-    pub fn new<T, L, S>(
-        logger: &Logger,
-        ethereum_adapter: Arc<T>,
-        link_resolver: Arc<L>,
-        store: Arc<S>,
+    pub fn new(
+        ethereum_adapter: Arc<dyn EthereumAdapter>,
```

Comment:
Yes, it wasn't being used anymore. More fundamentally, passing loggers to the methods that need them instead of storing them in the data structures seems like an approach that better preserves logging context.